### ISSUE TITLE
Create new user while Shout is running

### ISSUE THREAD

**Issue by furier** [erming/shout#362]
Let an administrator create a user while shout is running. 

Currently the shout instance goes down when invoking the `shout add <username>` command, causing all users currently connected to disconnect.
And then you have start up shout again...

**Comment by erming** [comment]
It crashes? That's a bug then. I'll check it out.

Basically, it _should_ work like this: when Shout is started, it loads every user.json in the `users/`-folder and then keep watching that folder for changes (add/remove).

**Comment by furier** [comment]
Yeah it stopped for me so all users got disconnected, then had to start shout again.

Btw what is the way you recommend to run shout as always on? Including on machine startup / reboots and if application simply crashes?

**Comment by erming** [comment]
I've not actually used it alot myself, but using `forever` seems like the best fit.

**Comment by furier** [comment]
If there are any logs where I can look for error messages or exceptions I could provide you those, because now i guess you are working blind... :S

**Comment by erming** [comment]
Nope, there's no log as long as no exception is thrown.

But here's the code: https://github.com/erming/shout/blob/master/src/clientManager.js#L114-L138

As you can see, every second it checks for added/removed user.json files and it will then iterate over them with `_.each()`. I guess you could add some `console.log()` in there to see what's going on.

**Comment by asabla** [comment]
It's working fine for me in a docker container

**Comment by tracker1** [comment]
@furier in addition to `forever` you may want to look at `pm2` as well.

**Comment by Ericson578** [comment]
does this look right for using forever with shout:

$forever start shout start

**Comment by furier** [comment]
@Ericson578 idk, but this works for me `$forever start /usr/bin/shout`

**Comment by furier** [comment]
Think you can close this one for now

**Comment by mayarichman** [comment]
I'd put the daemon suggestions in the documentation. For me what worked on my Debian was:
```
$ forever start /usr/local/lib/node_modules/shout start
```
